File: JAVA/KuantumKaosYoneticisiJava/src/models/KaranlikMadde.java
package models;

import interfaces.IKritik;

public class KaranlikMadde extends KuantumNesnesi implements IKritik {
    
    public KaranlikMadde(String id) {
        super(id, 7, 60); 
    }

    @Override
    public void AnalizEt() {
        azaltVeKontrolEt(15); // Stabilite 15 birim düşer.
        System.out.println("[KaranlikMadde - " + getID() + "] Radyasyon seviyesi yükseldi. Stabilite -15.");
    }

    @Override
    public void AcilDurumSogutmasi() {
        setStabilite(getStabilite() + 50); 
        System.out.println("[KaranlikMadde - " + getID() + "] Acil Soğutma Başarılı! Stabilite +50.");
    }
}

File: JAVASCRIPT/kuantum_kaos_yoneticisi_js/models/KaranlikMadde.js
// models/KaranlikMadde.js
const KuantumNesnesi = require('./KuantumNesnesi');

class KaranlikMadde extends KuantumNesnesi {
    constructor(id) {
        super(id, 7, 60); 
    }

    AnalizEt() {
        this._azaltVeKontrolEt(15);
        console.log(`[KaranlikMadde - ${this.ID}] Radyasyon seviyesi yükseldi. Stabilite -15.`);
    }

    // IKritik arayüz metodu
    AcilDurumSogutmasi() {
        this.stabilite += 50; 
        console.log(`[KaranlikMadde - ${this.ID}] Acil Soğutma Başarılı! Stabilite +50.`);
    }
}
module.exports = KaranlikMadde;